### PR TITLE
[privatebin] Add privatebin helm chart

### PR DESCRIPTION
**Pull request by psych0d0g**
**Description of the change**
 added parameterized helm chart to deploy PrivateBIN (a client encrypted self-hosted pastebin alternative)

**Benefits**

More deployable products ;)

**Possible drawbacks**

no known limitations yet

### CODE REVIEW

File: charts/incubator/privatebin/README_CHANGELOG.md.gotmpl [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{{- define "custom.changelog.header" -}}
```
Comment (bjw-s) [inline charts/incubator/privatebin/README_CHANGELOG.md.gotmpl:1]: This file is no longer required with the latest version of our template.

File: charts/incubator/privatebin/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- name: psych0d0g
+  email: psych0d0g@users.noreply.github.com
+dependencies:
+- name: common
+  repository: https://library-charts.k8s-at-home.com
+  version: 4.2.0
+- name: postgresql
+  version: 10.13.10
+  repository: https://charts.bitnami.com/bitnami
+  condition: postgresql.enabled
```
Comment (bjw-s) [inline charts/incubator/privatebin/Chart.yaml:25]: Please add an annotation to the chart with the changes. Check out [the template chart](https://github.com/k8s-at-home/charts/blob/master/templates/chart/Chart.yaml) for an example.

**Review by bjw-s — CHANGES_REQUESTED** [review]
Thanks for taking the time to create this PR! 🎖 I've left some comments.

Afterwards you can run `./hack/gen-helm-docs.sh stable <chart>` again to update the chart's `README.md` file.

If you have any questions, feel free to reach out!

**Comment by psych0d0g** [comment]
Running `./hack/gen-helm-docs.sh stable privatebin` leads to:
WARN[2022-01-07T08:36:45+01:00] Error generating gotemplates for chart /.../k8sathome/charts/stable/privatebin: template: /.../k8sathome/charts/stable/privatebin:120: function "fromYaml" not defined 

did i miss anything?

**Comment by psych0d0g** [comment]
> 

was fixed by using kah custom helm-docs version 0.1.1

File: charts/stable/privatebin/values.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  postgresqlUsername: privatebin
+  postgresqlPassword: privatebin
+  postgresqlDatabase: privatebin
+  persistence:
+    enabled: false
+    # storageClass: ""
+
+# -- Application Settings. See https://github.com/PrivateBin/PrivateBin/blob/master/cfg/conf.sample.php
+# for a description of every setting
+config:
```
Comment (bjw-s) [inline charts/stable/privatebin/values.yaml:59]:
```suggestion
# for a description of every setting
# @default -- See values.yaml
config:
```

File: charts/stable/privatebin/templates/common.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+{{- define "privatebin.configVolume" -}}
+enabled: "true"
+mountPath: "/srv/cfg/conf.php"
+subPath: "conf.php"
+type: "custom"
+volumeSpec:
+  configMap:
+    name: {{ template "common.names.fullname" . }}-config
+{{- end -}}
+{{- $_ := set .Values.persistence "privatebin-config" (include "privatebin.configVolume" . | fromYaml) -}}
```
Comment (bjw-s) [inline charts/stable/privatebin/templates/common.yaml:14]:
```suggestion
{{/* Append the hardcoded settings */}}
{{- define "privatebin.harcodedValues" -}}
  {{/* Append the configMap volume to the volumes */}}
  persistence:
    privatebin-config:
      enabled: true
      type: "configMap"
      name: "{{ include "common.names.fullname" . }}-config"
      mountPath: "/srv/cfg/conf.php"
      subPath: "conf.php"
  {{- end }}
{{- end -}}
{{- $_ := mergeOverwrite .Values (include "privatebin.harcodedValues" . | fromYaml) -}}
```

We have (sort of unofficially) moved away from the previous style of injecting "hardcoded" values and moved to the pattern as I've laid out in my suggestion.

File: charts/stable/privatebin/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+apiVersion: v2
+appVersion: 1.3.5
+description: PrivateBin running on an Nginx, php-fpm & Alpine Linux stack
```
Comment (bjw-s) [inline charts/stable/privatebin/Chart.yaml:3]:
```suggestion
description: A minimalist, open source online pastebin running on an Nginx, php-fpm & Alpine Linux stack
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
Looking good!! Just a few more changes requested, if you can get those in there I'll go ahead and approve/merge.

File: charts/stable/privatebin/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    - kind: fixed
+      description: set Initial version of the chart to 1.0.0
+    - kind: removed
+      description: removed obsoleted file README_CHANGELOG.md.gotmpl
+    - kind: changed
+      description: upstream postgresql chart updated to 10.14.4
+    - kind: fixed
+      description: moved application specific config to bottom of values.yaml
+    - kind: changed
+      description: bump common chart version from 4.2.0 to 4.3.0
```
Comment (bjw-s) [inline charts/stable/privatebin/Chart.yaml:41]:
```suggestion
    - kind: added
      description: Initial chart version
```

**Review by bjw-s — APPROVED** [review]
LGTM! :🍻

**Comment by bjw-s** [comment]
@all-contributors Please add @psych0d0g for code.

**Pull request merged**